spec/unit/knife/cookbook_metadata_spec.rb

```diff
@@ -174,6 +175,15 @@
         expect(@stderr.string).to match /#{description} blah/im
       end
     end
+    it "should raise a syntax error on invalid JSON" do
+      expect(File).to receive(:exist?).with("#{@cookbook_dir}/foobar/metadata.json").
+        and_return(true)
+      expect(IO).to receive(:read).with("#{@cookbook_dir}/foobar/metadata.json").
+        and_return(@invalid_json_data)
```

Comment:
Just as an update I've created some metadata.json files in the spec/data/metadata directory and kept some pressure on this over the last few days but I have a test that isn't throwing an exception, when it should. Need some time to figure out whats up there and I'll rebase & push another commit up.